packages/api-fetch/src/test/index.js

```diff
@@ -113,83 +113,86 @@ describe( 'apiFetch', () => {
 		);
 	} );
 
-	it( 'should return the error message properly', () => {
-		window.fetch.mockReturnValue(
-			Promise.resolve( {
-				status: 400,
-				json() {
-					return Promise.resolve( {
-						code: 'bad_request',
-						message: 'Bad Request',
-					} );
-				},
-			} )
-		);
+	it( 'should return the error message properly', async () => {
+		globalThis.fetch.mockResolvedValue( {
+			ok: false,
+			status: 400,
+			async json() {
+				return {
+					code: 'bad_request',
+					message: 'Bad Request',
+				};
+			},
+		} );
 
-		return apiFetch( { path: '/random' } ).catch( ( body ) => {
-			// eslint-disable-next-line jest/no-conditional-expect
-			expect( body ).toEqual( {
-				code: 'bad_request',
-				message: 'Bad Request',
-			} );
+		await expect( apiFetch( { path: '/random' } ) ).rejects.toEqual( {
+			code: 'bad_request',
+			message: 'Bad Request',
```

Comment:
This removes the need to disable `jest/no-conditional-expect`. And also the test checks that the promise really rejects and fails otherwise. In the original code, the `expect` inside the `catch` callback would never run, and the test would finish without calling any `expect`.